packages/noya-state/src/reducers/exportReducer.ts

```diff
@@ -0,0 +1,113 @@
+import Sketch from '@sketch-hq/sketch-file-format-ts';
+import produce from 'immer';
+import { getSelectedLayers } from '../selectors/layerSelectors';
+import { ApplicationState } from './applicationReducer';
+
+export type ExportAction =
+  | [type: 'setScale', index: number, value: string]
+  | [type: 'setName', index: number, value: string]
+  | [type: 'setFileFormat', index: number, value: Sketch.ExportFileFormat]
+  | [
+      type: 'setNamingScheme',
+      index: number,
+      value: Sketch.ExportFormatNamingScheme,
+    ]
+  | [type: 'addExportFormat']
+  | [type: 'deleteExportFormat', index: number];
+
+export function exportReducer(
+  state: ApplicationState,
+  action: ExportAction,
+): ApplicationState {
+  switch (action[0]) {
+    case 'setScale': {
+      const [, index, value] = action;
+
+      return produce(state, (draft) => {
+        const layers = getSelectedLayers(draft);
+
+        const number = isNaN(parseFloat(value))
+          ? parseFloat(value.slice(0, -1))
+          : parseFloat(value);
+
+        const visibleScale =
+          value.slice(-1) === 'w'
+            ? Sketch.VisibleScaleType.Width
+            : value.slice(-1) === 'h'
+            ? Sketch.VisibleScaleType.Height
+            : Sketch.VisibleScaleType.Scale;
```

Comment:
We should only be parsing/formatting this once. We currently have similar code both here and in the UI component. I think we could make a function that parses the input and returns an object, which we can then pass as the action `value`

```ts
type ExportScale = { scale: number, visibleScaleType: Sketch.VisibleScaleType }

function parseScale(scaleText: string): ExportScale | undefined {
  // ...
}
```

If the user types something invalid, then `parseScale` should return `undefined`, and we call `reset` on the input. It should never dispatch an action.

Reply:
Do add this in a selector ? 🤔

Reply:
I'd probably make a new file `noya-state/src/exportOptions.ts` and put it in there. There's a lot of stuff in selectors that we can probably move elsewhere at some point